components/core/src/os/users/admincheck.c

```diff
@@ -0,0 +1,90 @@
+#include <windows.h>
+#include <winnt.h>
+#include <ShlObj.h>
+
+#pragma comment (lib,"shell32.lib")
+
+TOKEN_ELEVATION_TYPE s_elevationType = TokenElevationTypeDefault;
+BOOL                 s_bIsAdmin = FALSE;
+
+// returns 0 if an admin
+// returns 1 if not
+int GetElevationState()
+{
+    if (GetProcessElevation(&s_elevationType, &s_bIsAdmin)) {
+        switch(s_elevationType) {
+            // Default user or UAC is disabled
+            case TokenElevationTypeDefault:  
+                if (IsUserAnAdmin()) {
+                    return 0;
```

Comment:
I updated the function to return different values based on the state of the user token, which will be useful in upcoming "process permission awareness" scenarios.   It still has the early returns, but has more types.  Again, this is one of those tech debt items that should be replaced with a "rusty" solution, but the win32 API doesn't make that easy.